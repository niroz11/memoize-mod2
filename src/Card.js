import React, { Component } from 'react';
class Card extends Component {
    constructor(){
        super()
        this.state = {
            clickedAnswer: ''
        }

    }
    handleClick = (event) => {
        
        this.setState({
          clickedAnswer: event.target.innerText
        })
      }
    
    render(){
        const answer = this.props.scopeData.answer
        console.log(answer, "answer")
        
      const option =  this.props.scopeData.options.map((option) => {
            return (
               <button onClick={this.handleClick}>{option}</button>
            )
        })
        return(
            <div>
            <h1>{this.props.scopeData.question}</h1>
            {option}
            </div>
        
        )
    }
}
export default Card;