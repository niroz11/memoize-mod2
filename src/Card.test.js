import React from "react";
import Card from "./Card";
import { shallow } from "enzyme";

let mockQuestion = 1;
let mockIncrement = jest.fn();
let mockScore = 1;
let mockUpdateScore = jest.fn();
let mockSaveLocalStorage = [
  {
    question: "What does Array.prototype.length returns?",
    options: [
      "length of an array",
      "length of string",
      "length of an object",
      "number of elements in array"
    ],
    answer: "number of elements in array",
    points: 10,
    key: 1
  },
  {
    question: "What does Array.prototype.pop() remove?",
    options: [
      "first element of an array",
      "last element of an array",
      "middle element of an array",
      "first and last element of an array"
    ],
    answer: "last element of an array",
    points: 10,
    key: 2
  }
];

describe("Card", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Card
        question={mockQuestion}
        increment={mockIncrement}
        score={mockScore}
        updateScore={mockUpdateScore}
        saveLocalStorage={mockSaveLocalStorage}
      />
    );
  });

  it("should match the snapshot with all data passed in", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
