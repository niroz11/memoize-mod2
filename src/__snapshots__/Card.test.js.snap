// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot with all data passed in 1`] = `undefined`;
